### DISCUSSION TITLE
20230828 update: JSON error generated when pressing Addons button

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1410] [Fixed Issues]
Running rAudio 20230620

JSON error is immediately generated when pressing the Addons button:

[Json Error.txt](https://github.com/rern/rAudio/files/12497950/Json.Error.txt)

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by hjfisch** [comment]
Same for me.
Did now a new install with the new build and use my backup .
It is working now for me without probs .

**Comment by voterpublic** [comment]
Same error as Sagnasty,
rAudio 20230630
6.1.28-5 aarch64
Raspberry Pi 3B Rev 1.2
HiFiBerry DAC+ Pro

**Comment by darts2020** [comment]
Same error as Sagnasty,
rAudio 20230630 (rAudio-RPi2-20230620 image)
Raspberry Pi 3B Rev 1.2
USB DAC

**Comment by Editpo** [comment]
Same error here.

rAudio 20230630
Raspberry Pi4
HiFiBerry DAC+ Pro

**Comment by xborg68** [comment]
Same 

Hifiberry AMP2 as DAC+

firmware-raspberrypi 20230619-1
linux-firmware 20230404.2e92a49f-1
linux-firmware-whence 20230404.2e92a49f-1
raspberrypi-bootloader 20230623-1
raspberrypi-firmware 20230516-1

**Comment by rern** [comment]
Please try: https://github.com/rern/rAudio/discussions/1421#discussioncomment-7037026

**Reply by voterpublic** [reply]
I tried but now get this,

```
Errors: No number after minus sign in JSON at position 2270 
{ "push" : { "bash" : "" , "url" : "" , "hide" : 1 } , "r1" : { "title" : "rAudio" , "version" : "20230918" , "revision" : [ "Fix - System - I2S Audio setting" , "Fix - Features - Browser on RPi" , "Fix - System - Audiophonics power switch" , "Improve - Switch from NGINX Pushstream to standard WebSocket" ] , "description" : "Updates for rAudio" , "nouninstall" : 1 , "thumbnail" : "/assets/img/coverart.svg" , "sourcecode" : "https://github.com/rern/rAudio" , "installurl" : "https://github.com/rern/rAudio/raw/main/install.sh" , "verify" : "[[ $( cat /srv/http/data/addons/r1 ) < 20230620 ]] && echo 1" , "notverified" : "Please download the latest image file to be able to update." } , "dab" : { "title" : "DAB Radio" , "description" : "Add Digital Audio Broadcasting radio reception capabilities(RTL-SDR dongle required)" , "thumbnail" : "/assets/img/addons/thumbdab.jpg" , "sourcecode" : "https://github.com/vale-max/rAudio-addons/raw/main/dabradio" , "installurl" : "https://github.com/rern/rAudio-addons/raw/main/dabradio/install.sh" , "hide" : "[[ -e /usr/bin/mediamtx || -e /boot/kernel.img ]] && echo 1" } , "fran" : { "title" : "Webradio - France Musique & Fip" , "buttonlabel" : [ "plus-circle", "Add" ] , "description" : "Webradio stations by France Musique and Fip" , "thumbnail" : "/assets/img/addons/thumbfran.gif" , "sourcecode" : "https://github.com/rern/rAudio-addons/raw/main/webradio" , "installurl" : "https://github.com/rern/rAudio-addons/raw/main/webradio/install.sh" , "hide" : "[[ -e /srv/http/data/webradio/Fip ]] && echo 1" } , "rank" : { "title" : "Rank Mirror Package Servers" , "description" : "Fix package download errors caused by unreachable servers.Rank mirror package servers by download speed and latency." , "thumbnail" : "/assets/img/addons/thumbrank.png" , "buttonlabel" : [ "bars", "Rank" ] , "sourcecode" : "https://github.com/rern/rAudio-addons/tree/main/rankmirrors" , "installurl" : "https://github.com/rern/rAudio-addons/raw/main/rankmirrors/rankmirrors.sh" , "option" : { "message" : "Download test for each server:" , "textlabel" : "Seconds" } , "hide" : "[[ -e /boot/kernel.img ]] && echo 1" }
	, "status" : {
		  "hide"        : [[ -Xe /boot/kernel.img ]] && echo 1]
		, "installed"   : ["r1"]
		, "notverified" : [""]
		, "update"      : ["r1"]
		, "online"      : true
	}
}```

thanks

**Reply by Sagnasty** [reply]
I performed the following commands via putty, but the contents of the common.sh file did not change after examining a file to file comparison with the original common.sh (same JSON error generated):
![image](https://github.com/rern/rAudio/assets/76575081/3281446f-f17a-4646-a239-badc1a3bbdf4)


I then tried deleting the common.sh file and doing a direct copied of the rAudio-20230918 common.sh file via WinSCP, with a warm start.  Resulting error: 
[New JSON error.txt](https://github.com/rern/rAudio/files/12685465/New.JSON.error.txt)

I then reverted back to the original common.sh file.

**Reply by voterpublic** [reply]
Hello,
Tried new install of version 'rAudio-64bit-20230918' - unable to get this to work on my system. (Raspberry Pi 3B Rev 1.2 + HiFiBerry DAC+ Pro)
No matter what I did it did not like the HiFiBerry DAC+ Pro, when option for DAC was applied in settings the options for the DAC failed to be shown in the MPD menu. Did this with various SD cards, fresh install no back up, fresh install with backup imported.
Nothing worked.

Tried with (fresh install) version 'rAudio-64bit-20230828'. Great success. Also able to restore from back up.
Update to latest version, via the addon menu, was successful (this took a little longer than any previous update with extra steps in the update process), system now displays,
rAudio 20230828
6.1.47-1 aarch64
Raspberry Pi 3B Rev 1.2

Many thanks Rern :)

**Reply by marcelgsteiger** [reply]
...I had exactly the same problem and found a workaround, see https://github.com/rern/rAudio/discussions/1430#discussioncomment-7165224

**Comment by Sagnasty** [comment]
No longer an issue in release 20231022 

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

Jim
